### PR TITLE
[BXMSPROD-1667] add stage-cpaas-artifacts job

### PR DESCRIPTION
**Pull request by rgdoliveira**
**Thank you for submitting this pull request**

**JIRA**: https://issues.redhat.com/browse/BXMSPROD-1667

<pre>
How to retest a PR or trigger a specific build:

* <b>a pull request</b> please add comment: <b>Jenkins retest this</b>
</pre>

### CODE REVIEW

File: job-dsls/src/main/resources/job-scripts/prod_rhba_stage_cpaas_artifacts.jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    }
+
+    post {
+        always {
+            cleanWs()
+        }
+    }
+}
+
+def createOfflinerZip(rhbaFolderPath) {
```
Comment (Ginxo) [inline job-dsls/src/main/resources/job-scripts/prod_rhba_stage_cpaas_artifacts.jenkinsfile:71]: either we move all the required global variables required by `createOfflinerZip` method like `RHBA_FOLDER_PATH`, `PRODUCT_NAME_LOWER_CASE`, `VERSION` etc... or we remove `rhbaFolderPath` argument and `RHBA_FOLDER_PATH` is directly used.

File: job-dsls/src/main/resources/job-scripts/prod_rhba_stage_cpaas_artifacts.jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    remoteExec("mkdir -p ${offlinerFolderPath}")
+    remoteExec("cp ${rhbaFolderPath}/offliner.txt ${offlinerFolderPath}")
+    sshagent(credentials: ['rcm-publish-server']) {
+        sh "scp -o StrictHostKeyChecking=no integration-platform-tooling/scripts/rhba/build-offline-repo.sh rhba@${RCM_HOST}:${offlinerFolderPath}/offline-repo-builder.sh"
+    }
+    remoteExec("cd ${rhbaFolderPath}; zip -r rhba-${VERSION}-offliner.zip ${offlinerFolderName}")
+    remoteExec("rm -rf ${offlinerFolderPath}")
+}
+
+def stageArtifact(artifact, rhbaFolderPath, targetFolderPath) {
```
Comment (Ginxo) [inline job-dsls/src/main/resources/job-scripts/prod_rhba_stage_cpaas_artifacts.jenkinsfile:85]: the same case here

File: job-dsls/src/main/resources/job-scripts/prod_rhba_stage_cpaas_artifacts.jenkinsfile [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    def targetArtifactPath = "${targetFolderPath}/${PRODUCT_NAME_LOWER_CASE}-${getMilestoneVersion(VERSION)}-${artifact.finalClassifier}"
+    println "${artifact.info} from ${artifactPath} to ${targetArtifactPath}"
+    remoteExec("${artifact.stageCommand} ${artifactPath} ${targetArtifactPath}")
+}
+
+def getMilestoneVersion(version) {
+    return version.substring(0, version.lastIndexOf("."))
+}
+
+def remoteExec(command) {
```
Comment (Ginxo) [inline job-dsls/src/main/resources/job-scripts/prod_rhba_stage_cpaas_artifacts.jenkinsfile:96]: the same case here

File: job-dsls/src/main/resources/job-scripts/prod_rhba_stage_cpaas_artifacts.jenkinsfile [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+    post {
+        always {
+            cleanWs()
+        }
+    }
+}
+
+def createOfflinerZip(rhbaFolderPath) {
+    println "Creating offliner zip"
```
Comment (Ginxo) [inline job-dsls/src/main/resources/job-scripts/prod_rhba_stage_cpaas_artifacts.jenkinsfile:72]:
```suggestion
    println 'Creating offliner zip'
```

File: job-dsls/src/main/resources/job-scripts/prod_rhba_stage_cpaas_artifacts.jenkinsfile [hunk 5]
```diff
@@ -0,0 +1,100 @@
+def artifacts = [
+    [originClassifier: "maven-repository.zip", finalClassifier: "maven-repository.zip", stageCommand: "cp", info: "Copying maven repository zip"],
```
Comment (Ginxo) [inline job-dsls/src/main/resources/job-scripts/prod_rhba_stage_cpaas_artifacts.jenkinsfile:2]:
```suggestion
    [originClassifier: 'maven-repository.zip', finalClassifier: 'maven-repository.zip', stageCommand: 'cp', info: 'Copying maven repository zip'],
```

File: job-dsls/src/main/resources/job-scripts/prod_rhba_stage_cpaas_artifacts.jenkinsfile [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+    [originClassifier: "license.zip", finalClassifier: "licenses.zip", stageCommand: "cp", info: "Copying licenses zip"],
```
Comment (Ginxo) [inline job-dsls/src/main/resources/job-scripts/prod_rhba_stage_cpaas_artifacts.jenkinsfile:3]:
```suggestion
    [originClassifier: 'license.zip', finalClassifier: 'licenses.zip', stageCommand: 'cp', info: 'Copying licenses zip'],
```

File: job-dsls/src/main/resources/job-scripts/prod_rhba_stage_cpaas_artifacts.jenkinsfile [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+    [originClassifier: "src.zip", finalClassifier: "sources.zip", stageCommand: "cp", info: "Copying sources zip"],
```
Comment (Ginxo) [inline job-dsls/src/main/resources/job-scripts/prod_rhba_stage_cpaas_artifacts.jenkinsfile:4]:
```suggestion
    [originClassifier: 'src.zip', finalClassifier: 'sources.zip', stageCommand: 'cp', info: 'Copying sources zip'],
```

File: job-dsls/src/main/resources/job-scripts/prod_rhba_stage_cpaas_artifacts.jenkinsfile [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+            }
+        }
+
+        stage('Clone integration-platform-tooling repository') {
+            steps {
+                sh "git clone ${INTEGRATION_PLATFORM_TOOLING_REPOSITORY} integration-platform-tooling"
+            }
+        }
+
+        stage("Create offliner zip") {
```
Comment (Ginxo) [inline job-dsls/src/main/resources/job-scripts/prod_rhba_stage_cpaas_artifacts.jenkinsfile:45]:
```suggestion
        stage('Create offliner zip') {
```

File: job-dsls/src/main/resources/job-scripts/prod_rhba_stage_cpaas_artifacts.jenkinsfile [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+
+def stageArtifact(artifact, rhbaFolderPath, targetFolderPath) {
+    def artifactPath = "${rhbaFolderPath}/rhba-${VERSION}-${artifact.originClassifier}"
+    def targetArtifactPath = "${targetFolderPath}/${PRODUCT_NAME_LOWER_CASE}-${getMilestoneVersion(VERSION)}-${artifact.finalClassifier}"
+    println "${artifact.info} from ${artifactPath} to ${targetArtifactPath}"
+    remoteExec("${artifact.stageCommand} ${artifactPath} ${targetArtifactPath}")
+}
+
+def getMilestoneVersion(version) {
+    return version.substring(0, version.lastIndexOf("."))
```
Comment (Ginxo) [inline job-dsls/src/main/resources/job-scripts/prod_rhba_stage_cpaas_artifacts.jenkinsfile:93]:
```suggestion
    return version.substring(0, version.lastIndexOf('.'))
```

**Review by Ginxo — COMMENTED** [review]
just few technical suggestions

**Review by AntStephenson — APPROVED** [review]

**Pull request merged**